### PR TITLE
NAS-113963 / 22.12 / Allow moving system dataset to a passphrase-encrypted pool (system dataset will be unencrypted)

### PR DESCRIPTION
**Pull request by themylogin**

### CODE REVIEW

**Comment by themylogin** [comment]
time 5:45

**Comment by pkgdemon** [comment]
retest this please

**Comment by pkgdemon** [comment]
retest this please

**Comment by themylogin** [comment]
@pkgdemon are tests ran for the branch or for the branch merged/rebase on top of master? If the former then the test might fail in a weird way (middleware killing itself). I just rebased this branch on top of master, should not fail any more.

File: src/middlewared/middlewared/plugins/sysdataset.py [hunk 1]
```diff
@@ -162,43 +161,26 @@ async def do_update(self, job, data):
                 if error := await self.destination_pool_error(new['pool']):
                     verrors.add('sysdataset_update.pool', error)
 
-        if new['pool'] and new['pool'] != await self.middleware.call('boot.pool_name'):
-            pool = await self.middleware.call('pool.query', [['name', '=', new['pool']]])
-            if not pool:
+        if new['pool'] == await self.middleware.call('boot.pool_name'):
```
Comment (sonicaj) [inline src/middlewared/middlewared/plugins/sysdataset.py:164]: Do we need this conditional ? I think `pool_choices` would still work nicely if the new configured pool is `boot-pool` ?

**Review by sonicaj — APPROVED** [review]

**Review by yocalebo — CHANGES_REQUESTED** [review]
Has this been tested on an HA system? We prevented encrypted system datasets from being used because of many factors.

**Comment by anodos325** [comment]
Has this been tested on server joined to AD?

**Comment by themylogin** [comment]
> Has this been tested on an HA system? We prevented encrypted system datasets from being used because of many factors.

Currently we don't prevent encrypted system datasets from being used. Also this PR also does not introduce encrypted system dataset, it will have `encryption=off`

**Comment by themylogin** [comment]
> Has this been tested on server joined to AD?

IIRC we don't allow changing system dataset is server is joined to AD, so I am not sure what should be tested?

**Comment by themylogin** [comment]
@yocalebo @anodos325 ping

**Comment by themylogin** [comment]
@anodos325 ping

**Review by anodos325 — APPROVED** [review]

**Comment by themylogin** [comment]
@yocalebo ping

**Comment by themylogin** [comment]
@yocalebo ping

**Review by yocalebo — APPROVED** [review]

**Pull request merged**